### PR TITLE
Add doxygen code examples for mqtt.h

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
Adds example code blocks to nontrivial API functions in mqtt.h

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/include/mqtt.h [hunk 1]
```diff
@@ -220,6 +220,48 @@ struct MQTTDeserializedInfo
  *
  * @return #MQTTBadParameter if invalid parameters are passed;
  * #MQTTSuccess otherwise.
+ *
+ * <b>Example</b>
+ * @code{c}
+ *
+ * //Function for obtaining a timestamp
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:227]: nit: space after the //?

File: libraries/standard/mqtt/include/mqtt.h [hunk 2]
```diff
@@ -278,6 +320,52 @@ MQTTStatus_t MQTT_Init( MQTTContext_t * pContext,
  *    the API makes one more network receive call in an attempt to receive the remaining
  *    2 bytes. In the worst case, it can happen that the remaining 2 bytes are never
  *    received and this API will end up spending timeoutMs + transport receive timeout.
+ *
+ * <b>Example</b>
+ * @code{c}
+ * 
+ * //variables used in this example
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:327]: Capital V?

File: libraries/standard/mqtt/include/mqtt.h [hunk 3]
```diff
@@ -278,6 +320,52 @@ MQTTStatus_t MQTT_Init( MQTTContext_t * pContext,
...
+ * //This is assumed to have been initialized before calling this function.
+ * MQTTContext_t * pContext;
+ *
+ * //True for creating a new session with broker, false if we want to resume an old one.
+ * connectInfo.cleanSession = true;
+ * //Client ID must be unique to broker. This field is required
+ * connectInfo.pClientIdentifier = "someClientID";
+ * connectInfo.clientIdentifierLength = strlen( connectInfo.pClientIdentifier );
+ * 
+ * //The following field are optional
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:341]: fields

File: libraries/standard/mqtt/include/mqtt.h [hunk 4]
```diff
@@ -278,6 +320,52 @@ MQTTStatus_t MQTT_Init( MQTTContext_t * pContext,
...
+ *
+ * //Send the connect packet. Use 100 ms as the timeout to wait for the CONNACK packet.
+ * status = MQTT_Connect( pContext, &connectInfo, &willInfo, 100, &sessionPresent );
+ * 
+ * if( status == MQTTSuccess )
+ * {
+ *      //Since we requested a clean session, this must be false
+ *      assert( sessionPresent == false );
+ * 
+ *      //do something with the connection.
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:366]: Capital D

File: libraries/standard/mqtt/include/mqtt.h [hunk 5]
```diff
@@ -299,6 +387,40 @@ MQTTStatus_t MQTT_Connect( MQTTContext_t * pContext,
...
+ * @code{c}
+ *
+ * //Variables used in this example
+ * MQTTStatus_t status;
+ * MQTTSubscribeInfo_t subscriptionList[ NUMBER_OF_SUBSCRIPTIONS ] = { 0 };
+ * uint16_t packetId;
+ * //This is assumed to have been initialized and connected before calling this function.
+ * MQTTContext_t * pContext;
+ * //This is assumed to be a list of filters we want to subscribe to.
+ * const char * filter[ NUMBER_OF_SUBSCRIPTIONS ];
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:401]: filters

File: libraries/standard/mqtt/include/mqtt.h [hunk 6]
```diff
@@ -316,6 +438,35 @@ MQTTStatus_t MQTT_Subscribe( MQTTContext_t * pContext,
...
  * #MQTTSuccess otherwise.
+ * 
+ * <b>Example</b>
+ * @code{c}
+ * 
+ * //Variables used in this example
+ * MQTTStatus_t status;
+ * MQTTPublishInfo_t publishInfo;
+ * uint16_t packetId;
+ * //This is assumed to be initialized and connected;
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:449]: extra semi-colon?

File: libraries/standard/mqtt/include/mqtt.h [hunk 7]
```diff
@@ -347,6 +498,40 @@ MQTTStatus_t MQTT_Ping( MQTTContext_t * pContext );
...
+ * @code{c}
+ *
+ * //Variables used in this example
+ * MQTTStatus_t status;
+ * MQTTSubscribeInfo_t unsubscribeList[ NUMBER_OF_SUBSCRIPTIONS ] = { 0 };
+ * uint16_t packetId;
+ * //This is assumed to have been initialized and connected before calling this function.
+ * MQTTContext_t * pContext;
+ * //This is assumed to be a list of filters we want to unsubscribe from.
+ * const char * filter[ NUMBER_OF_SUBSCRIPTIONS ];
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:512]: filters

File: libraries/standard/mqtt/include/mqtt.h [hunk 8]
```diff
@@ -347,6 +498,40 @@ MQTTStatus_t MQTT_Ping( MQTTContext_t * pContext );
...
+ * 
+ * //Set information for each unsubscription request
+ * for( int i = 0; i < NUMBER_OF_SUBSCRIPTIONS; i++ )
+ * {
+ *      //Each subscription needs a topic filter,
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt.h:517]: unsubscription

Comment (sarenameas) [reply libraries/standard/mqtt/include/mqtt.h:517]: wait is this a word? lol

Comment (sarenameas) [reply libraries/standard/mqtt/include/mqtt.h:517]: Maybe we can just use unsubscribe?

File: libraries/standard/mqtt/include/mqtt.h (hunk 1 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Added

File: libraries/standard/mqtt/include/mqtt.h (hunk 3 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Thanks

File: libraries/standard/mqtt/include/mqtt.h (hunk 8 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Thanks, it's not. I changed each instance to "unsubscribe request" or "unsubscribing"

File: libraries/standard/mqtt/include/mqtt.h (hunk 6 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Thanks

File: libraries/standard/mqtt/include/mqtt.h (hunk 4 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Changed

File: libraries/standard/mqtt/include/mqtt.h (hunk 3 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Thanks

File: libraries/standard/mqtt/include/mqtt.h (hunk 2 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h]: Changed

File: libraries/standard/mqtt/include/mqtt.h [hunk 9]
```diff
@@ -383,6 +567,30 @@ MQTTStatus_t MQTT_Disconnect( MQTTContext_t * pContext );
...
+ * // This context is assumed to be initialized and connected.
+ * MQTTContext_t * pContext;
+ * 
+ * while( true )
+ * {
+ *      status = MQTT_ProcessLoop( pContext, timeoutMs );
+ * 
+ *      if( status != MQTTSuccess )
+ *      {
+ *          // Determine the error. It's possible we might need to disconnect TCP.
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt.h:586]: It's possible we might need to disconnect TCP --> It's possible we might need to disconnect the underlying transport connection.

File: libraries/standard/mqtt/include/mqtt.h [hunk 10]
```diff
@@ -220,6 +220,48 @@ struct MQTTDeserializedInfo
...
+ * <b>Example</b>
+ * @code{c}
+ *
+ * // Function for obtaining a timestamp.
+ * uint32_t getTimeStampMs();
+ * // Callback function for receiving packets.
+ * void eventCallback(
+ *      MQTTContext_t * pContext,
+ *      MQTTPacketInfo_t * pPacketInfo,
+ *      MQTTDeserializedInfo_t * pDeserialized
```
[the remark this answers is not in the document]
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h:232]: Thanks, will change in next doc PR

File: libraries/standard/mqtt/include/mqtt.h (hunk 9 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt.h:585]: Thanks, will update the wording in next PR

**Pull request merged**